File: leetcode/136_single_number.cpp
// Copyright (c) 2018 Jacopo Notarstefano

#include <cassert>
#include <vector>

using namespace std;

class Solution {
 public:
    int singleNumber(vector<int>& nums) {
        int result = 0;

        for (auto num : nums) {
            result ^= num;
        }

        return result;
    }
};

int main() {
    auto solution = Solution();

    vector<int> nums = {2, 2, 1};

    assert(1 == solution.singleNumber(nums));
}

File: leetcode/136_single_number.py
# -*- coding: utf-8 -*-

from __future__ import absolute_import, division, print_function


class Solution(object):
    def singleNumber(self, nums):
        result = 0

        for num in nums:
            result ^= num

        return result


if __name__ == '__main__':
    solution = Solution()

    assert 1 == solution.singleNumber([2, 2, 1])
    assert 4 == solution.singleNumber([4, 1, 2, 1, 2])
